lib/diag.c

```diff
@@ -752,14 +855,98 @@ int switchtec_diag_port_eq_tx_coeff(struct switchtec_dev *dev, int port_id,
 }
 
 /**
- * @brief Get the far end TX equalization table
+ * @brief Get the port equalization TX coefficients
+ * @param[in]  dev	Switchtec device handle
+ * @param[in]  port_id	Physical port ID
+ * @param[in]  end      Get coefficents for the Local or the Far End
+ * @param[out] res      Resulting port equalization coefficients
+ *
+ * @return 0 on success, error code on failure
+ */
+int switchtec_diag_port_eq_tx_coeff(struct switchtec_dev *dev, int port_id,
+		enum switchtec_diag_end end, enum switchtec_diag_link link,
+		struct switchtec_port_eq_coeff *res)
+{
+	int ret = -1;
+
+	if (switchtec_is_gen5(dev)) {
```

Comment:
Style nit: for single statement following if/else, the surrounding should not be present.